WordPress/Classes/Models/Post.swift

```diff
@@ -195,7 +195,7 @@ class Post: AbstractPost {
 
         if let originalPost = original as? Post {
 
-            if tags?.characters.count != originalPost.tags?.characters.count && tags != originalPost.tags {
+            if tags ?? "" != originalPost.tags ?? "" {
```

Comment:
> I would like to avoid having 10+ functions to compare 10+ fields and a big || for all of them

Why so?  This code...

```swift
override func hasLocalChanges() -> Bool {
    if super.hasLocalChanges() {
        return true
    }
 
    assert(original == nil || original is Post)
        
    if let originalPost = original as? Post,
        tagsHaveLocalChanges()
        || coordsHaveLocalChanges() 
        || publicizeMessageHasLocalChanges() {

        return true
    }

    return false
}
```

...has the advantage of being self-documenting when compared to the current direction, which is:

```swift
override func hasLocalChanges() -> Bool {
    if super.hasLocalChanges() {
        return true
    }

    assert(original == nil || original is Post)

    if let originalPost = original as? Post {

        if tags ?? "" != originalPost.tags ?? "" {
            return true
        }

        if let coord1 = geolocation?.coordinate,
            let coord2 = originalPost.geolocation?.coordinate, coord1.latitude != coord2.latitude || coord1.longitude != coord2.longitude {

            return true
        }

        if publicizeMessage ?? "" != originalPost.publicizeMessage ?? "" {
            return true
        }

        if (!NSDictionary(dictionary: disabledPublicizeConnections ?? [:])
            .isEqual(to: originalPost.disabledPublicizeConnections ?? [:])) {
            return true
        }
    }

    return false
}
```

> Furthermore the functions would only be used here.

That's not the only reason to separate code in methods, though.  Abstraction from details, single responsibility and separation of concerns are also valid reasons to separate functionality (one such example is when the application starts up... most of the code there is executed only once, but it's still separated in steps).

That said, this PR doesn't need to be more than a bug fix for the time being and your changes are perfectly fine in regards to that.

I'm approving the PR.